core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -181,11 +181,13 @@ object Controller {
         val instance = if (args.length > 0) args(1).toInt else 0
 
         // initialize the runtimes manifest
-        if (config.isValid && ExecManifest.initialize(config)) {
+        val execManifest = ExecManifest.initialize(config)
+        if (config.isValid && execManifest.isSuccess) {
             val port = config.servicePort.toInt
             BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
         } else {
-            logger.error(this, "Bad configuration, cannot start.")
+            logger.error(this, "Bad configuration, cannot start." +
+              (if (execManifest.isFailure) " Invalid runtimes.manifest: " + execManifest.failed.get.toString else ""))
```

Comment:
Done.

> Note that pulling the exec manifest initialization to precede the configuration check changes the behavior on which error is caught/reported first. 

Right. Now we see the errors like this:
```
[2017-05-18T20:13:57.743Z] [ERROR] [??] [Controller] Required property runtimes.manifest is invalid: java.util.NoSuchElementException: key not found: runtimes 
[2017-05-18T20:13:57.743Z] [ERROR] [??] [Config] required property whisk.version.buildno still not set
```

If there's a preference on the order, let me know.